content/redirect.js

```diff
@@ -11,7 +11,37 @@ const {
 
 const FORBIDDEN_URL_SYMBOLS = ["\n", "\t"];
 
+// Module-level cache
 let ARCHIVED_URLS;
+let ARCHIVED_PATHS;
+
+function getArchivedURLs() {
+  if (!ARCHIVED_URLS) {
+    ARCHIVED_URLS = new Set();
+    for (const folder of getArchivedPaths()) {
+      // We don't need to use `path.sep` here because all the files in the
+      // archived.txt file are written explicitly with `/`.
+      const split = folder.split("/");
+      split.unshift("");
+      split.splice(2, 0, "docs");
```

Comment:
paths with a `_colon_` cannot be safely reverted to an URL. Either we hack this in here an are well aware of it not being 100% accurate or we stick with converting urls to paths and compare those.

Reply:
Good catch! I can't believe I missed that. Yeah, we need the reverse of slugToFoldername (or whatever it's called).

Reply:
Ugh. That reminds me, we don't have a reverse of `slugToFolder` :(
I could write a function that does:
```
   .replace(/_doublecolon_/g, '::')
```
etc.
But that doesn't take care of whatever `sanitizeFilename` might do. That one is not mapping 1-to-1. 
Also, we'd lose the casing. But that might not be a great solution. 
What we should perhaps do is to rewrite `archived.txt` to be a list of URLs instead. 
What do you think about that?
Last time I wrote a python script that traversed all the `index.html` files in mdn/archived-content and extracted the slug.

Reply:
We chatted about it on Zoom and concluded that the simplest solution is to do this:
```diff
function isArchivedURL(url) {
- return getArchivedURLs().has(url)
+ return getArchivedFilepaths().has(slugToFoldername(url))  
}
```